db.py

```diff
@@ -10,19 +19,67 @@ def get_db_connection():
         cursor = conn.cursor(cursor_factory=RealDictCursor)
         return cursor, conn
     except Exception as e:
-        print(f"Failed to connect to DB.")
+        logging.error("Failed to connect to DB.", exc_info=True)
         return None, None
     
 def get_user_by_email(email):
     try:
         cursor, conn = get_db_connection()
+        if cursor is None or conn is None:
+            raise DatabaseConnectionError("No connection to DB.")
+            logging.debug("No connection to the database.")
+            return None
+
         cursor.execute("SELECT * FROM users WHERE email = %s", (email,))
-        user = cursor.fetchone() # return dict - with properties as keys
+        user = cursor.fetchone()
+        if not user:
+            raise UserNotFoundError(f"No user found with email:{email}")
         cursor.close()
         conn.close()
         return user
+    
+    except OperationalError as e:
+        logging.error(f"An error occured when trying to fetch user.", exc_info=True)
+        raise DatabaseConnectionError("Failed to connect to the db")
+
+    except DatabaseError as e:
+            logging.error("Database error:", exc_info=True)
+            raise DatabaseConnectionError("An error occurred while fetching user.")
+        
     except Exception as e:
-        print(f"Error fetching user by email: {email}")
+        logging.debug(f"Error fetching user by email: {email}", exc_info=True)
         return None
 
+
+def check_is_valid_user_id(user_id):
```

Comment:
Why just user_id and not the whole user object? You are basically making a SELECT call to the DB that cost you, and you probably will need to use other stuff and not just the id, so you'll do another SELECT in a different layer?